### PR TITLE
Ic2 compat config

### PR DESCRIPTION
**Pull request by maggi373**
add config for ic2 tier support for mekanism
change tier for mekanism machines and generators to tier 2 from 4
add a fix from leytera's fork of MK:CE, did a code search after having a bug with the cable when testing and found this, didnt fix the issue i had tho.
also set rf as default energy measurement

### CODE REVIEW

File: src/main/java/mekanism/common/CommonProxy.java [hunk 1]
```diff
@@ -311,8 +311,10 @@ public void loadConfiguration()
 		general.OreDictPlatinum = Mekanism.configuration.get(Configuration.CATEGORY_GENERAL, "OreDictPlatinum", false).getBoolean();
 		general.enableSiliconCompat = Mekanism.configuration.get(Configuration.CATEGORY_GENERAL, "EnableSiliconCompat", true).getBoolean();
 
+		general.IC2SinkTier = Mekanism.configuration.get(Configuration.CATEGORY_GENERAL, "IC2SinkTier", 4, "Tier level for machines, we recommend setting it to tier 2 for ic2 classic", 1, 4).getInt();
+		general.IC2SourceTier = Mekanism.configuration.get(Configuration.CATEGORY_GENERAL, "IC2SourceTier", 4, "Tier level for generators, we recommend setting it to tier 2 for ic2 classic", 1, 4).getInt();
 
-		String s = Mekanism.configuration.get(Configuration.CATEGORY_GENERAL, "EnergyType", "J", null, new String[]{"J", "RF", "MJ", "EU"}).getString();
```
Comment (DrParadox7) [inline src/main/java/mekanism/common/CommonProxy.java:316]: Why is Joules getting removed in an IC2 oriented update?

Comment (maggi373) [reply src/main/java/mekanism/common/CommonProxy.java:316]: default is changed to rf, i forgot that i was on this branch when committing.

Comment (DrParadox7) [reply src/main/java/mekanism/common/CommonProxy.java:316]: Is the config still parsing power related entries as Joules?

Comment (maggi373) [reply src/main/java/mekanism/common/CommonProxy.java:316]: yes, i just changed the default setting to rf from j, all of them still works. changed it due to the large amount of modpacks using rf and changing to rf will bring some cohesion in the experience. Note that this setting only changes the shown values in-game

**Review by DrParadox7 — APPROVED** [review]

**Pull request merged**